image.h

```diff
@@ -5,11 +5,25 @@
 class Image {
 public:
 	Image(uint16_t rows, uint16_t columns): m_rows(rows), m_columns(columns) {
-		pixels = new uint8_t[m_rows * m_columns];
+		if ((m_columns <= 1024 && m_rows <= 1024)) 
+		{
+			//Allocate memory only if its in range
+			pixels = new uint8_t[m_rows * m_columns];
+		}
+		else
+		{
+			pixels = nullptr;
```

Comment:
Nice handling of pixels and nullptr - even in the destructor :)
But - the behavior is a bit implicit - if the image is invalid, the pixels become nullptr. This is not evident, unless the consumers of the class see its implementation. Why not throw an exception?